hushline/settings.py

```diff
@@ -457,8 +384,53 @@ def verify_2fa_setup() -> Response | str:
         session.pop("is_setting_up_2fa", None)
         return redirect(url_for("logout"))
 
-    @bp.route("/update_pgp_key", methods=["GET", "POST"])
-    @require_2fa
+    @bp.route("/update_pgp_key_proton", methods=["POST"])
+    @authentication_required
+    def update_pgp_key_proton() -> Response | str:  # noqa: PLR0911
+        user_id = session.get("user_id")
+        if not user_id:
+            flash("⛔️ User not authenticated.")
+            return redirect(url_for("login"))
+
+        user = db.session.get(User, user_id)
+        form = PGPProtonForm()
+        if form.validate_on_submit():
+            email = form.email.data
+
+            if is_valid_email(email):
```

Comment:
Let's move this into a form validator?

Reply:
Resolved in https://github.com/scidsg/hushline/pull/491/commits/8931a8aa54f6c532af7665696e789f7263e19573